components/raftstore/src/store/worker/read.rs

```diff
@@ -207,13 +213,61 @@ where
     }
 }
 
-/// #[RaftstoreCommon]: LocalReadContext combines some LocalReader's fields for temporary usage.
 pub struct LocalReadContext<'a, E>
 where
     E: KvEngine,
 {
-    read_id: &'a mut ThreadReadId,
-    snap_cache: &'a mut Box<Option<Arc<E::Snapshot>>>,
+    read_id: Option<ThreadReadId>,
+    snap_cache: &'a mut SnapCache<E>,
+}
+
+impl<'a, E> LocalReadContext<'a, E>
+where
+    E: KvEngine,
+{
+    fn new(snap_cache: &'a mut SnapCache<E>, read_id: Option<ThreadReadId>) -> Self {
+        Self {
+            snap_cache,
+            read_id,
+        }
+    }
+
+    // Update the snapshot in the `snap_cache` if the read_id is None or does not
+    // match. When the read_id is None, it ok to just update the snapshot without
+    // touching the `cached_read_id` because only a consecutive requests will
+    // have the same read_id, which means this `cached_read_id` will not be matched
```

Comment:
> which means this `cached_read_id` will not be matched anymore

If that's the case, then clear `cached_read_id` seems also OK.

Reply:
Fine